stdlib/src/builtin/string.mojo

```diff
@@ -149,23 +149,28 @@ fn chr(c: Int) -> String:
 # ===----------------------------------------------------------------------===#
 
 
-# TODO: this is hard coded for decimal base
 @always_inline
-fn _atol(str_ref: StringRef) raises -> Int:
-    """Parses the given string as a base-10 integer and returns that value.
+fn _atol(str_ref: StringRef, base: Int = 10) raises -> Int:
+    """Parses the given string as an integer in the given base and returns that value.
 
     For example, `atol("19")` returns `19`. If the given string cannot be parsed
     as an integer value, an error is raised. For example, `atol("hi")` raises an
     error.
 
     Args:
-        str_ref: A string to be parsed as a base-10 integer.
+        str_ref: A string to be parsed as an integer in the given base.
+        base: Base used for conversion, value must be between 2 and 36.
 
     Returns:
         An integer value that represents the string, or otherwise raises.
     """
     if not str_ref:
         raise Error("Empty String cannot be converted to integer.")
+    if base < 2 or base > 36:
```

Comment:
**Question** Any specific reason to exclude the special case of base `0`?  In Python:

```python
int("0", 1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: int() base must be >= 2 and <= 36, or 0
```

which suggests we should handle base 0 too pedantically.

Reply:
No reason, I didn't realize that base 0 was an option, I guess I'll handle this case too